careerfairsystems/nexpo-web
# Slow migration

Essential packages updated and some put on hold.

# Details

```
$ yarn outdated                                                                    
yarn outdated v1.22.4
info Color legend : 
 "<red>"    : Major Update backward-incompatible updates 
 "<yellow>" : Minor Update backward-compatible features 
 "<green>"  : Patch Update backward-compatible bug fixes
color    Package     Current Wanted  Latest  Package Type           URL                                        
<red>    antd        3.26.18 3.26.18 4.4.1   dependencies           https://ant.design/                        
<red>    flow-bin    0.84.0  0.84.0  0.128.0 devDependencies        https://github.com/flowtype/flow-bin#readme
<red>    react-redux 6.0.1   6.0.1   7.2.0   dependencies           https://github.com/reduxjs/react-redux     
<yellow> semver      7.1.3   7.1.3   7.3.2   resolutionDependencies https://github.com/npm/node-semver#readme
```

`antd`: migration guide https://ant.design/docs/react/migration-v4
`flow-bin`: Huge changelog. Do a bisect and see how far it is possible to upgrade without breakage.
`react-redux`: tests fail when updating to latest. Will require some (minor?) code changes to upgrade.
`semver`: see #1a87e97

# To verify
`yarn outdated` should preferably show the same output as above.
`yarn test`: all should pass
`yarn audit` should not report any serious vulnerabilities.

`yarn start` and check manually.

# Side effects
I think the actual build process is much slower. Old dependencies somewhere? Investigate why?
